File: common/src/main/java/net/impleri/playerskills/server/NetHandler.java
package net.impleri.playerskills.server;

import net.impleri.playerskills.PlayerSkills;
import net.impleri.playerskills.network.SyncSkillsMessage;
import net.impleri.playerskills.server.events.SkillChangedEvent;
import net.minecraft.server.level.ServerPlayer;

public abstract class NetHandler {
    public static void syncPlayer(SkillChangedEvent<?> event) {
        var player = event.getPlayer();

        if (player instanceof ServerPlayer serverPlayer) {
            syncPlayer(serverPlayer);
        } else {
            PlayerSkills.LOGGER.warn("Attempted to sync skill changes from clientside");
        }
    }

    public static void syncPlayer(ServerPlayer player) {
        if (player == null) {
            PlayerSkills.LOGGER.warn("Attempted to sync skill changes for nobody");
            return;
        }

        var skills = ServerApi.getAllSkills(player);
        PlayerSkills.LOGGER.debug("Syncing {} player skills to {}", skills.size(), player.getName().getString());

        new SyncSkillsMessage(player, skills).sendTo(player);
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/server/NetHandler.kt
package net.impleri.playerskills.server

import net.impleri.playerskills.api.Player
import net.impleri.playerskills.events.SkillChangedEvent
import net.impleri.playerskills.network.SyncSkillsMessage
import net.impleri.playerskills.utils.PlayerSkillsLogger
import net.minecraft.server.level.ServerPlayer

object NetHandler {
  fun syncPlayer(event: SkillChangedEvent<*>) {
    val player = event.player
    if (player is ServerPlayer) {
      syncPlayer(player, false)
    } else {
      PlayerSkillsLogger.SKILLS.warn("Attempted to sync skill changes from clientside")
    }
  }

  @JvmOverloads
  fun syncPlayer(player: ServerPlayer?, force: Boolean = true) {
    if (player == null) {
      PlayerSkillsLogger.SKILLS.warn("Attempted to sync skill changes for nobody")
      return
    }
    val skills = Player.get(player).toMutableList()
    PlayerSkillsLogger.SKILLS.debug("Syncing ${skills.size} player skills to ${player.name}")
    SyncSkillsMessage(player, skills, force).sendTo(player)
  }
}
